apps/server/src/database/prisma/repositories/prisma-locations-repository.ts

```diff
@@ -30,15 +30,41 @@ export class PrismaLocationsRepository implements ILocationsRepository {
 
     async deleteMany(locationsIds: string[]): Promise<void> {
         try {
-            await prisma.location.deleteMany({
+            for (const locationId of locationsIds) {
+            const location = await prisma.location.findUnique({
+              where: {
+                id: locationId,
+              },
+              include: {
+                subLocation: true,
+              },
+            })
+      
+            if (!location) {
+              throw new PrismaError('Repository Error: Location not found')
+            }
+      
+            if (location.subLocation.length > 0) {
```

Comment:
I think it will be better and easier to use `DELETE CASCADE` intead of for looping to delete the sublocations